### ISSUE TITLE
LongIdGenerator is not properly supported

### ISSUE THREAD

**Issue by amcghie** [ExpediaDotCom/haystack-client-java#105]
The `LongIdGenerator` was introduced in commit #b93f6685 however there are still numerous places in the code that assume that `traceId` is a `UUID`:

For example in the `TextMapPropagator#extract(TextMap carrier)` method the following code exists (which assumes the `traceId`, `spanId` and `parentId` are all `UUID`'s):
```
        SpanContext context = new SpanContext(UUID.fromString(traceId),
                                              UUID.fromString(spanId),
                                              parentId == null ? null : UUID.fromString(parentId),
                                              true);
```
When the `LongIdGenerator` is used (in a client) the following error occurs:
```
Error handling a request: 9747e8d6c092bd0b
! java.lang.IllegalArgumentException: Invalid UUID string: 1560033070049436460
! at java.util.UUID.fromString(UUID.java:194)
! at com.expedia.www.haystack.client.propagation.TextMapPropagator.extract(TextMapPropagator.java:101)
! at com.expedia.www.haystack.client.propagation.TextMapPropagator.extract(TextMapPropagator.java:30)
! at com.expedia.www.haystack.client.Tracer.extract(Tracer.java:185)
! at com.expedia.www.haystack.client.Tracer.extract(Tracer.java:43)
! at io.opentracing.contrib.jaxrs2.server.ServerTracingFilter.parentSpanContext(ServerTracingFilter.java:99)
! at io.opentracing.contrib.jaxrs2.server.ServerTracingFilter.filter(ServerTracingFilter.java:68)
! at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:132)
<snip>
```
This can be easily replicated by running the example in https://github.com/ExpediaDotCom/haystack-dropwizard-example

There is also a reference to `UUID.randomUUID()` on Line 327 in Tracer.java which should be replaced by `tracer.idGenerator.generate()`

**Comment by amcghie** [comment]
Is it possible to get a rough idea of a release timeframe for 0.2.9?

**Comment by nsahai8** [comment]
The issue is fixed in release 0.2.9
